src/main/java/seedu/address/model/profile/UniqueProfileItemList.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.model.profile;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * A list of profile items that enforces uniqueness between its elements and does not allow nulls.
+ * A profile item is considered unique by comparing using {@code Profile#isSameProfileItem(ProfileItem)}. As such,
+ * adding and updating of profile items uses ProfileItem#isSameProfileItem(ProfileItem) for equality so as to ensure
+ * that the profile item being added or updated is unique in terms of identity in the UniquePersonList.
+ * However, the removal of a profile item uses ProfileItem#equals(Object) so as to ensure that the profile item with
+ * exactly the same  fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see ProfileItem#isSameProfileItem(ProfileItem)
+ */
+public class UniqueProfileItemList implements Iterable<ProfileItem> {
+
+    private final ObservableList<ProfileItem> internalList = FXCollections.observableArrayList();
+    private final ObservableList<ProfileItem> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent person as the given argument.
+     */
+    public boolean contains(ProfileItem toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameProfileItem);
+    }
+
+    /**
+     * Adds a profileItem to the list.
+     * The person must not already exist in the list.
+     */
+    public void add(ProfileItem toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicatePersonException();
```

Comment:
wrong exceptions in this file or are they placeholders?

Reply:
I have not created exceptions yet so this is a placeholder